src/dpl/src/Graphics.h

```diff
@@ -32,35 +32,33 @@
 
 namespace dpl {
 
-using odb::dbBlock;
-using odb::dbInst;
-using odb::Point;
-using odb::Rect;
-
 class Opendp;
 struct Cell;
 
+// Decorates the gui::Renderer with DPL specific routines.
 class Graphics : public gui::Renderer
 {
  public:
-  Graphics(Opendp* dp, float min_displacement, const dbInst* debug_instance);
+  Graphics(Opendp* dp, float min_displacement, const odb::dbInst* debug_instance);
 
-  void startPlacement(dbBlock* block);
-  void placeInstance(dbInst* instance);
-  void binSearch(const Cell* cell, int xl, int yl, int xh, int yh);
-  void endPlacement();
+  virtual void startPlacement(odb::dbBlock* block);
+  virtual void placeInstance(odb::dbInst* instance);
+  virtual void binSearch(const Cell* cell, int xl, int yl, int xh, int yh);
+  virtual void endPlacement();
```

Comment:
Why are you making these virtual?

Reply:
Their implementations rely on Graphics which relies on gui which relies on OpenROAD symbols. By making these virtual we avoid having to do symbol resolution.

A more complete solution would be to decouple graphics from dpl, and introduce observers for these methods instead. I wanted to try to keep it simple to start.